src/CommitmentsV2/SFA.DAS.CommitmentsV2/Application/Commands/UpdateAccountLegalEntityName/UpdateAccountLegalEntityNameCommandHandler.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using SFA.DAS.CommitmentsV2.Data;
+
+namespace SFA.DAS.CommitmentsV2.Application.Commands.UpdateAccountLegalEntityName
+{
+    public class UpdateAccountLegalEntityNameCommandHandler : AsyncRequestHandler<UpdateAccountLegalEntityNameCommand>
+    {
+        private readonly Lazy<AccountsDbContext> _db;
+
+        public UpdateAccountLegalEntityNameCommandHandler(Lazy<AccountsDbContext> db)
+        {
+            _db = db;
+        }
+
+        protected override async Task Handle(UpdateAccountLegalEntityNameCommand request, CancellationToken cancellationToken)
+        {
+
+            var accountLegalEntity = await _db.Value.AccountLegalEntities.IgnoreQueryFilters().SingleAsync(a => a.Id == request.AccountLegalEntityId, cancellationToken);
```

Comment:
I'm not sure what I think of the model-level queries. The effect of the IgnoreQueryFilters() here is to include deleted account legal entities. This isn't obvious from looking at the code. Am happy to see what team says and go along with consensus though.

Reply:
It's to handle an update of soft deleted AccountLegalEntity. Otherwise the out of sequence message will end up on the errors queue. In our situation we have no real reason to have this filter at the moment (well in this DbContext)